### ISSUE TITLE
dcraw crash on Fuji Finepix HS10 with address sanitizer.

### ISSUE THREAD

**Issue by hfiguiere** [Beep6581/RawTherapee#6708]
**Short description**
dcraw cause address sanitizer errors with Fuji Finepix HS10 files.

**Steps to reproduce**

Download the RAW file from https://raw.pixls.us/getfile.php/1652/nice/Fujifilm%20-%20FinePix%20HS10%20HS11%20-%2012bit%20uncompressed%20(4:3).RAF  

1. Build rawtherapee with `-DWITH_ASAN=address`
2. Go to the directory with the file above
3. Double click on the file

Crash. See below for the output.

**Expected behavior**

Doesn't crash.

**Additional information**

- Version of RawTherapee: `dev` branch. Reproduced in 5.9 or commit ca08a279d50b55d278ca1b6659a8892fac44d663
- Version of your operating system: Linux Fedora 37
- If you built from source, the contents of the `AboutThisBuild.txt` file in the installation folder:
```
Version: 5.9-126-gca08a279d
Branch: dev
Commit: ca08a279d
Commit date: 2023-03-10
Compiler: cc 12.2.1
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.7
Lensfun: V0.3.3.0
Build type: Debug
Build flags:  -std=c++11 -ffp-contract=off  -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -fsanitize=address -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -g -ftree-vectorize
Link flags:   -fsanitize=address
OpenMP support: ON
MMAP support: ON
Build OS: Linux 6.1.14-200.fc37.x86_64 x86_64
Build date: Mon, 13 Mar 2023 18:28:38 +0000 UTC
Build epoch: 1678732118
Build UUID: 5a5319d8-e2be-4d9b-b304-5701a5c354fc
```

Other useful information:

This is clear cut case a `strcpy` misuse. The documentation explicit says that source and dest shall not overlap.

The address sanitizer output:
```
=================================================================
==461684==ERROR: AddressSanitizer: strcpy-param-overlap: memory ranges [0x7f038d5e0b9d,0x7f038d5e0ba7) and [0x7f038d5e0ba5, 0x7f038d5e0baf) overlap
    #0 0x7f03b7a61fff in __interceptor_strcpy.part.0 (/lib64/libasan.so.8+0x61fff)
    #1 0x24c2af3 in DCraw::identify() /home/hub/source/RawTherapee/rtengine/dcraw.cc:9810
    #2 0x2230ed5 in rtengine::RawImage::loadRaw(bool, unsigned int, bool, rtengine::ProgressListener*, double) /home/hub/source/RawTherapee/rtengine/rawimage.cc:470
    #3 0x22ebc33 in rtengine::Thumbnail::loadQuickFromRaw(Glib::ustring const&, rtengine::RawMetaDataLocation&, rtengine::SensorType&, int&, int&, int, bool, bool, bool) /home/hub/source/RawTherapee/rtengine/rtthumbnail.cc:406
    #4 0x162f5f0 in Thumbnail::_generateThumbnailImage() /home/hub/source/RawTherapee/rtgui/thumbnail.cc:171
    #5 0x162e79a in Thumbnail::Thumbnail(CacheManager*, Glib::ustring const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/hub/source/RawTherapee/rtgui/thumbnail.cc:108
    #6 0x5860a8 in CacheManager::getEntry(Glib::ustring const&) /home/hub/source/RawTherapee/rtgui/cachemanager.cc:120
    #7 0x141be2e in PreviewLoader::Impl::processNextJob() /home/hub/source/RawTherapee/rtgui/previewloader.cc:130
    #8 0x141f1c1 in sigc::bound_mem_functor0<void, PreviewLoader::Impl>::operator()() const /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
    #9 0x141ef33 in sigc::adaptor_functor<sigc::bound_mem_functor0<void, PreviewLoader::Impl> >::operator()() const /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
    #10 0x141ead8 in sigc::internal::slot_call0<sigc::bound_mem_functor0<void, PreviewLoader::Impl>, void>::call_it(sigc::internal::slot_rep*) /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
    #11 0x7f03b65ee119 in (anonymous namespace)::call_thread_entry_slot(void*, void*) (/lib64/libglibmm-2.4.so.1+0x67119)
    #12 0x7f03b6ac8ab1 in g_thread_pool_thread_proxy.lto_priv.0 (/lib64/libglib-2.0.so.0+0x85ab1)
    #13 0x7f03b6ac3981 in g_thread_proxy (/lib64/libglib-2.0.so.0+0x80981)
    #14 0x7f03b4cae12c in start_thread (/lib64/libc.so.6+0x8b12c)
    #15 0x7f03b4d2fbbf in __clone3 (/lib64/libc.so.6+0x10cbbf)

0x7f038d5e0b9d is located 925 bytes inside of 205912-byte region [0x7f038d5e0800,0x7f038d612c58)
allocated by thread T77 here:
    #0 0x7f03b7abb1a8 in operator new(unsigned long) (/lib64/libasan.so.8+0xbb1a8)
    #1 0x22ebbd6 in rtengine::Thumbnail::loadQuickFromRaw(Glib::ustring const&, rtengine::RawMetaDataLocation&, rtengine::SensorType&, int&, int&, int, bool, bool, bool) /home/hub/source/RawTherapee/rtengine/rtthumbnail.cc:404
    #2 0x162f5f0 in Thumbnail::_generateThumbnailImage() /home/hub/source/RawTherapee/rtgui/thumbnail.cc:171
    #3 0x162e79a in Thumbnail::Thumbnail(CacheManager*, Glib::ustring const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/hub/source/RawTherapee/rtgui/thumbnail.cc:108
    #4 0x5860a8 in CacheManager::getEntry(Glib::ustring const&) /home/hub/source/RawTherapee/rtgui/cachemanager.cc:120
    #5 0x141be2e in PreviewLoader::Impl::processNextJob() /home/hub/source/RawTherapee/rtgui/previewloader.cc:130
    #6 0x141f1c1 in sigc::bound_mem_functor0<void, PreviewLoader::Impl>::operator()() const /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
    #7 0x141ef33 in sigc::adaptor_functor<sigc::bound_mem_functor0<void, PreviewLoader::Impl> >::operator()() const /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
    #8 0x141ead8 in sigc::internal::slot_call0<sigc::bound_mem_functor0<void, PreviewLoader::Impl>, void>::call_it(sigc::internal::slot_rep*) /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
    #9 0x7f03b65ee119 in (anonymous namespace)::call_thread_entry_slot(void*, void*) (/lib64/libglibmm-2.4.so.1+0x67119)

Thread T77 created by T0 here:
    #0 0x7f03b7a4b3e6 in __interceptor_pthread_create (/lib64/libasan.so.8+0x4b3e6)
    #1 0x7f03b6ac6b20 in g_thread_new_internal (/lib64/libglib-2.0.so.0+0x83b20)

0x7f038d5e0ba5 is located 933 bytes inside of 205912-byte region [0x7f038d5e0800,0x7f038d612c58)
allocated by thread T77 here:
    #0 0x7f03b7abb1a8 in operator new(unsigned long) (/lib64/libasan.so.8+0xbb1a8)
    #1 0x22ebbd6 in rtengine::Thumbnail::loadQuickFromRaw(Glib::ustring const&, rtengine::RawMetaDataLocation&, rtengine::SensorType&, int&, int&, int, bool, bool, bool) /home/hub/source/RawTherapee/rtengine/rtthumbnail.cc:404
    #2 0x162f5f0 in Thumbnail::_generateThumbnailImage() /home/hub/source/RawTherapee/rtgui/thumbnail.cc:171
    #3 0x162e79a in Thumbnail::Thumbnail(CacheManager*, Glib::ustring const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/hub/source/RawTherapee/rtgui/thumbnail.cc:108
    #4 0x5860a8 in CacheManager::getEntry(Glib::ustring const&) /home/hub/source/RawTherapee/rtgui/cachemanager.cc:120
    #5 0x141be2e in PreviewLoader::Impl::processNextJob() /home/hub/source/RawTherapee/rtgui/previewloader.cc:130
    #6 0x141f1c1 in sigc::bound_mem_functor0<void, PreviewLoader::Impl>::operator()() const /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
    #7 0x141ef33 in sigc::adaptor_functor<sigc::bound_mem_functor0<void, PreviewLoader::Impl> >::operator()() const /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
    #8 0x141ead8 in sigc::internal::slot_call0<sigc::bound_mem_functor0<void, PreviewLoader::Impl>, void>::call_it(sigc::internal::slot_rep*) /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
    #9 0x7f03b65ee119 in (anonymous namespace)::call_thread_entry_slot(void*, void*) (/lib64/libglibmm-2.4.so.1+0x67119)

SUMMARY: AddressSanitizer: strcpy-param-overlap (/lib64/libasan.so.8+0x61fff) in __interceptor_strcpy.part.0
==461684==ABORTING
```

**Comment by hfiguiere** [comment]
See PR #6709

**Comment by Lawrence37** [comment]
Resolved with PR #6709.
